scripts/shellcheck.sh

```diff
@@ -0,0 +1,243 @@
+    printf "%d\n" "$CORE_COUNT"
+}
+
+show_shellcheck_args() {
+    echo -e "Running shellcheck with severity level '${CYAN}${SEVERITY}${RST_COLOR}'"
+    if [[ -n "$EXCLUDE_CODES" ]]; then
+        echo
+        echo -e "Excluding error codes: '${YELLOW}${EXCLUDE_CODES}${RST_COLOR}'"
+    fi
+    if [[ -n "$INCLUDE_CODES" ]]; then
+        echo
+        echo -e "Including error codes: '${YELLOW}${INCLUDE_CODES}${RST_COLOR}'"
+    fi
+    echo
+}
+
+run_shellcheck() {
+    # two runs of shellcheck used since shellcheck doesn't allow for include codes
+    # to be checked at all levels while using a severity level for all other codes
+    #
+    # 1. include codes only all severity levels (if non-empty)
+    # 2. exclude codes with the include codes added at the entered severity level
+    RUN_1_STATUS=0
+    RUN_2_STATUS=0
+
+    # this prevents duplicate scanning of the include codes on the excludes run
+    EXCLUDES_ALL="$EXCLUDE_CODES"
+    if [[ -n "$INCLUDE_CODES" && -n "$EXCLUDE_CODES" ]]; then
+        EXCLUDES_ALL+=",${INCLUDE_CODES}"
+    elif [[ -n "$INCLUDE_CODES" ]]; then
+        EXCLUDES_ALL="${INCLUDE_CODES}"
+    fi
+
+    if command -v git &>/dev/null; then
+        SHELL_SCRIPTS=$(git ls-files '*.sh')
+    else
+        # exclude 3rdparty scripts by specifying the scripts directory
+        SHELL_SCRIPTS=$(find "$ROOT"/scripts -name '*.sh')
+    fi
+
+    # scan for only include codes, at all severity levels
+    if [[ -n "$INCLUDE_CODES" ]]; then
+        xargs -n 1 -P "$NUM_CORES" shellcheck -x -s bash -C"$COLOR" --include="$INCLUDE_CODES" -S "style" <<< "$SHELL_SCRIPTS"
+        RUN_1_STATUS="$?"
+    fi
+
+    # blanket severity level scan, omitting any include codes (to not repeat what shellcheck just scanned for)
+    xargs -n 1 -P "$NUM_CORES" shellcheck -x -s bash -C"$COLOR" --exclude="$EXCLUDES_ALL" -S "$SEVERITY" <<< "$SHELL_SCRIPTS"
+    RUN_2_STATUS="$?"
+
+    if [[ "$RUN_1_STATUS" -eq 0 && "$RUN_2_STATUS" -eq 0 ]]; then
+        echo -e "${GREEN}[PASS]${RST_COLOR} Shellcheck did not detect violations"
+        return 0
+    else
+        echo -e "${RED}[FAIL]${RST_COLOR} Shellcheck found violations"
+        return 1
+    fi
+}
+
+print_final_message() {
+    if [[ "$#" -ne 1 ]] || [[ "$1" -ne 0 && "$1" -ne 1 ]]; then
+        exit_on_error "This function needs a decimal status code as input"
+    fi
+
+    EXIT_STATUS="$1"
+    # if no errors found given our parameters, then shellcheck passed
+    if [[ "$EXIT_STATUS" -eq 0 ]]; then
+        echo -e "${GREEN}[PASS]${RST_COLOR} Shellcheck did not detect violations"
+        echo; echo -e "${GREEN}Shellcheck passed.${RST_COLOR}"; echo
+    else
+        echo -e "${RED}[FAIL]${RST_COLOR} Shellcheck found unexcused violations"
+        echo; echo -e "${RED}Shellcheck failed. Please fix the issues and try again.${RST_COLOR}"
+    fi
+    exit "$EXIT_STATUS"
+}
+
+main() {
+    check_shellcheck_install
+    parse_cli_args "$@"
+
+    if [[ "$COLOR" != "never" ]]; then
```

Comment:
This is fine, but `auto` should be removed from the help output, it's not being handled (which is okay! handling auto-coloring is a bit sticky).